pkg/cloud/services/compute/instance.go

```diff
@@ -98,6 +98,14 @@ func (s *Service) InstanceCreate(clusterName string, machine *clusterv1.Machine,
 	if err != nil {
 		return nil, err
 	}
+	if openStackCluster.Spec.ManagedSecurityGroups {
+		if util.IsControlPlaneMachine(machine) {
+			securityGroups = append(securityGroups, openStackCluster.Status.ControlPlaneSecurityGroup.ID)
+		} else {
+			securityGroups = append(securityGroups, openStackCluster.Status.WorkerSecurityGroup.ID)
```

Comment:
so we are allowing both sec group from Machine spec and managed security groups created
maybe follow PR update doc will also talk about this to make it more clearer to end user

Reply:
Yes, we are. Still we can add more security groups if we need.

Reply:
This PR also leads to a behavior where managesecuritygroups: true makes it impossible to not use the created sec groups. But I guess that's okay as all that managesecuritygroups does is to create these groups. Wouldn't make much sense to set the flag if they shouldn't be used.